src/main/java/oliver/Oliver.java

```diff
@@ -0,0 +1,205 @@
+package oliver;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Oliver {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    private boolean isRunning = true;
+    private final static String FILEPATH = "./data/oliver.txt";
+
+    public Oliver(String filePath) {
+        this.ui = new Ui();
+        this.storage = new Storage(filePath);
+        this.storage.initialise();
+        try {
+            this.tasks = new TaskList(storage.loadData());
+        } catch (IOException e) {
+            System.out.println("Error occurred when reading data into list.");
+            this.tasks = new TaskList();
+        }
+    }
+
+    public void run() {
+        System.out.println(ui.showWelcome());
+        while (this.isRunning) {
+            String input = ui.readInput();
+            System.out.println(getResponse(input));
+        }
+    }
+
+    public String getResponse(String input) {
+        String command = Parser.parseCommand(input);
+        if (input.equalsIgnoreCase("bye")) {
+            storage.writeToFile(this.tasks);
+            ui.close();
+            this.isRunning = false;
+            return ui.showBye();
+        } else if (input.equalsIgnoreCase("list")) {
+            return ui.showList(this.tasks);
+        } else if (command.equalsIgnoreCase("mark")) {
+            return handleMark(input);
+        } else if (command.equalsIgnoreCase("unmark")) {
+            return handleUnmark(input);
+        } else if (command.equalsIgnoreCase("todo")) {
+            return handleTodo(input);
+        } else if (command.equalsIgnoreCase("deadline")) {
+            return handleDeadline(input);
+        } else if (command.equalsIgnoreCase("event")) {
+            return handleEvent(input);
+        } else if (command.equalsIgnoreCase("delete")) {
+            return handleDelete(input);
+        } else if (command.equalsIgnoreCase("find")) {
+            return ui.showSearch(this.tasks, Parser.parseArgs(input));
+        } else {
+            return "Invalid command. Command was not recognised.";
+        }
+    }
```

Comment:
Can a switch statement or enum be used to handle these if-else statements to make your code cleaner?